include/flatbuffers/flatbuffers.h

```diff
@@ -685,6 +709,16 @@ FLATBUFFERS_FINAL_CLASS
     if (string_pool) delete string_pool;
   }
 
+  void Reset() {
```

Comment:
It's not entirely the same when there's an existing buffer. `Clear` reuses, and `Reset` deallocates (with a lazy allocate when it's first used next). This is an important distinction with the gRPC slices, where it's not safe to reuse the buffers.